### PR TITLE
UML-726 Behat smoke test

### PR DESCRIPTION
**Pull request by cooperaj**
## Purpose
This PR changes our smoke test suite from behave to behat. It utilises CircleCI containers with browsers embedded to run behat tests under chrome. 

## Approach

It's possible to run the tests locally by attaching the docker-compose.testing.yml file to your list of compose files and running 'smoke-test'

## Learning

Had to write a behat extension to get chrome and xvfb running as a part of the suite.

## Checklist

* [x] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [x] I have added tests to prove my work
* [ ] The product team have tested these changes

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-version: "2.1"
+version: "2.4"
```
Comment (williamfalconeruk) [inline docker-compose.yml:1]: we've had some issues with this and the gateway before on up having a version mismatch. I guess this is ok in terms of versions, as it's a minor version?

Comment (cooperaj) [reply docker-compose.yml:1]: no. quite right. 2.4 is needed to get phpstorms running of the tests working. i've made local changes to my api-gateway to have this work.

BUT

I think i've now worked around the issue so this isn't needed.

Still, probably should upgrade the version at some point right.

File: tests/smoke/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+```shell script
+# ensure your use-an-lpa environment is running already
+# use PHPStorm or a method devised in the top level README
+
+# run tests when wanted
+$ composer behat
+
+# create feature step definitions 
+$ composer behat -- --snippets-for
```
Comment (williamfalconeruk) [inline tests/smoke/README.md:12]: so this creates the step definitions (i guess as empty functions) - that's good to know 👍

Comment (cooperaj) [reply tests/smoke/README.md:12]: it spits them out on the console at the end of the run so you can copy and paste into a context file

You can of course always use PHPstorm to do it.

**Comment by cooperaj** [comment]
Moving to a different PR

**Pull request closed without merging**